### ISSUE TITLE
Writing to disk till 0 bytes left

### ISSUE THREAD

**Issue by an0n666** [arvidn/libtorrent#4730]
Libtorrent writes to disk till it becomes full(if file size > available storage). And when that happens, you can't even move the files to another drive. It fails with this message:

`6/7/2020 10:53 AM - Could not move torrent: xyz storage move failed. file_rename (F:\xyz.mkv): There is not enough space on the disk`

Also since the move failed, the save location should be reverted to whatever it was previously. Otherwise you end up with new save location with no files at all.

There has to be some space left on the drive to be able to move the files to another drive.
I'm not sure if it's a shortcoming of the client itself or this library. Hope this can be addressed someway.

**Comment by arvidn** [comment]
If you try to move the torrent, and it fails, the save path should remain unchanged. It sounds like you're saying it isn't, that sounds like a bug.

can you say something more about how you observe the save path changing? Do you see new files being downloaded to the new path?

internall, both the disk thread(s) and the network thread sore the save path. when moving a torrent, a message is sent to the disk threads which will update the path (if successful) and when the response comes back to the network thread it will also update the path (if the move was successful). At least that's the intention.

So, perhaps there is a way for these to get out of sync.

**Comment by an0n666** [comment]
This is the warning message when the disk ran out of space:
`(W) 2020-06-07T10:53:00 - File error alert. Torrent: "xyz". File: "xyz.mkv". Reason: xyz torrent file_write (F:\xyz.mkv) error: There is not enough space on the disk`

Me trying to move the file to another disk to complete the downloading:

`(N) 2020-06-07T10:53:25 - Set location: moving "xyz", from "F:\" to "H:\"`
`(C) 2020-06-07T10:53:26 - Could not move torrent: xyz. Reason: xyz storage move failed. file_rename (xyz.mkv): There is not enough space on the disk`

Since the move failed, I try again...and it shows this:

`(N) 2020-06-07T11:00:49 - Set location: moving "xyz", from "H:\" to "H:\"`

So this confirms that the location was already changed to H:\ from F:\ at this point of time without moving the actual files.

**Comment by an0n666** [comment]
BTW this torrent had multiple subfolders and files inside them. And I was partially leeching it. Donno if that has any significance...

To correct the issue I then closed qBt and manually moved the files to H:\ and started qBt. I ran a recheck and after it was done, the torrent would never start downloading for some weird reason. I then removed and re-added the torrent and it started downloading after the automatic recheck.
